File: backend/application-management/src/main/java/com/cybersoft/application_management/service/AuthService.java
package com.cybersoft.application_management.service;

import com.cybersoft.application_management.dto.request.LoginRequest;
import com.cybersoft.application_management.dto.request.RefreshTokenRequest;
import com.cybersoft.application_management.dto.request.RegisterRequest;
import com.cybersoft.application_management.dto.response.LoginResponse;

public interface AuthService {
    LoginResponse register(RegisterRequest request);
    LoginResponse login(LoginRequest request);
    LoginResponse refreshToken(RefreshTokenRequest request);
    void logout(RefreshTokenRequest request);

}

File: frontend/src/features/auth/api/auth-service.ts
import type {
  LoginRequest,
  LoginResponse,
  RefreshTokenRequest,
  RegisterRequest,
} from "../../../api/generated";

import authenticationApi from "./auth-api";

export async function login(
  request: LoginRequest,
): Promise<LoginResponse> {
  const response = await authenticationApi.login(request);

  return response.data;
}

export async function register(
  request: RegisterRequest,
): Promise<LoginResponse> {
  const response = await authenticationApi.register(request);

  return response.data;
}

export async function refreshToken(
  request: RefreshTokenRequest,
): Promise<LoginResponse> {
  const response =
      await authenticationApi.refreshToken(request);

  return response.data;
}

export async function logout(
  request: RefreshTokenRequest,
): Promise<void> {
  await authenticationApi.logout(request);
}